p2p/src/simulated/metrics.rs

```diff
@@ -0,0 +1,60 @@
+use commonware_cryptography::{utils::hex, PublicKey};
+use prometheus_client::encoding::EncodeLabelSet;
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+pub struct ReceivedMessage {
+    pub peer: String,
+    pub message: i32,
+}
+
+impl ReceivedMessage {
+    const UNKNOWN_TYPE: i32 = -1;
+    const TOO_LARGE_TYPE: i32 = -2;
+
+    pub fn new_unknown(peer: &PublicKey) -> Self {
+        Self {
+        peer: hex(peer),
+        message: Self::UNKNOWN_TYPE,
+        }
+    }
+
+    pub fn new_too_large(peer: &PublicKey) -> Self {
+        Self {
+        peer: hex(peer),
+        message: Self::TOO_LARGE_TYPE,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+pub struct SentMessage {
```

Comment:
Likewise, you should just record channel here. In this one, you won't need the `UNKNOWN` const because you can only send over channels that you've registered.